linked_list/linked_list.py

```diff
@@ -13,47 +13,63 @@ def __init__(self):
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1) accessing it only on
+    # Space Complexity: O(1) no new data structure being created
     def get_first(self):
-        pass
+        if self.head == None:
+            return None
 
+        return self.head.value
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: o(1)
+    # Space Complexity: o(1) constant time, since new data structure is being created but its small
     def add_first(self, value):
-        pass
-
+        self.head = Node(value,self.head)
+      
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     # Time Complexity: ?
     # Space Complexity: ?
     def search(self, value):
-        pass
+        current = self.head
+        while current is not None:
+            if value == current.value:
+                return True
+            current =  current.next 
+# reset current to equal the next node in the list
+        return False 
 
     # method that returns the length of the singly linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: 0(1) constant time
+    # Space Complexity: 0(1)
     def length(self):
-        pass
+        current = self.head
+        length = 0
+
+        while current != None:
+            length += 1
+            current = current.next 
+
+            return length

```

Comment:
Move this line out of the while loop; otherwise, it will return immediately after the first iteration:

```suggestion
        return length
```